internal/handler/cmd_query.go

```diff
@@ -22,15 +22,45 @@ import (
 	"github.com/FerretDB/FerretDB/internal/handler/common"
 	"github.com/FerretDB/FerretDB/internal/handler/handlererrors"
 	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 	"github.com/FerretDB/FerretDB/internal/util/must"
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
 // CmdQuery implements deprecated OP_QUERY message handling.
 func (h *Handler) CmdQuery(ctx context.Context, query *wire.OpQuery) (*wire.OpReply, error) {
-	cmd := query.Query().Command()
+	q := query.Query()
+	cmd := q.Command()
 	collection := query.FullCollectionName
 
+	v, _ := q.Get("speculativeAuthenticate")
```

Comment:
I see codecov doesn't complain about this line being uncovered, so it means that `speculativeAuthenticate` is used somewhere. Do we see where it's used? Is it used by the driver under the hood?

Reply:
Yes exactly, see https://github.com/FerretDB/FerretDB/pull/4111#discussion_r1505376061